lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -230,25 +230,41 @@ def inspect_all
     end
 
     def tagged_with?(category, name)
-      object_send(:is_tagged_with?, name.to_s, :ns => "/managed/#{category}")
+      if taggable?
+        object_send(:is_tagged_with?, name.to_s, :ns => "/managed/#{category}")
+      else
+        raise MiqAeException::UntaggableModel, "Model #{self.class} doesn't support tagging!"
+      end
     end
 
     def tags(category = nil)
-      ns = category.nil? ? "/managed" : "/managed/#{category}"
-      object_send(:tag_list, :ns => ns).split
+      if taggable?
+        ns = category.nil? ? "/managed" : "/managed/#{category}"
+        object_send(:tag_list, :ns => ns).split
+      else
+        raise MiqAeException::UntaggableModel, "Model #{self.class} doesn't support tagging!"
+      end
     end
 
     def tag_assign(tag)
-      ar_method do
-        Classification.classify_by_tag(@object, "/managed/#{tag}")
-        true
+      if taggable?
+        ar_method do
+          Classification.classify_by_tag(@object, "/managed/#{tag}")
+          true
+        end
+      else
+        raise MiqAeException::UntaggableModel, "Model #{self.class} doesn't support tagging!"
       end
     end
 
     def tag_unassign(tag)
-      ar_method do
-        Classification.unclassify_by_tag(@object, "/managed/#{tag}")
-        true
+      if taggable?
+        ar_method do
+          Classification.unclassify_by_tag(@object, "/managed/#{tag}")
+          true
+        end
+      else
+        raise MiqAeException::UntaggableModel, "Model #{self.class} doesn't support tagging!"
```

Comment:
@pkomanek You can DRY up this logic by creating a private method to check and raise an error if tagging is not supported by the class and just call that method at the top of each of the tagging methods.

```ruby
def verify_taggable_model
  raise MiqAeException::UntaggableModel, "Model #{self.class} doesn't support tagging" unless taggable?
end
private :verify_taggable_model
```

Then the methods just add one line:
```ruby
def tagged_with?(category, name)
  verify_taggable_model
  object_send(:is_tagged_with?, name.to_s, :ns => "/managed/#{category}")
end
```